docs/src/test-advanced-js.md

```diff
@@ -301,6 +301,58 @@ test('test', async ({ page }) => {
 });
 ```
 
+You can make arbitrary data available in your tests from your global setup file by setting them as environment variables via `process.env`.
+
+```js js-flavor=js
+// global-setup.js
+module.exports = async config => {
+  process.env.FOO = 'some data';
+  // Or a more complicated data structure as JSON:
+  process.env.BAR = JSON.stringify({ some: 'data' });
+};
+```
+
+```js js-flavor=ts
+// global-setup.ts
+import { FullConfig } from '@playwright/test';
+
+async function globalSetup(config: FullConfig) {
+  process.env.FOO = 'some data';
+  // Or a more complicated data structure as JSON:
+  process.env.BAR = JSON.stringify({ some: 'data' });
+}
+
+export default globalSetup;
+```
+
+Tests have access to the `process.env` properties set in the global setup.
+
+```js js-flavor=ts
+import { test } from '@playwright/test';
+const { FOO, BAR } = process.env;
```

Comment:
can we inline these? This way it'll be clear that these can be used inside the test, but not outside to declare tests